Unstructured-IO/pipeline-sec-filings
# feat: sync version

Similar to https://github.com/Unstructured-IO/unstructured/pull/25 but expanded.

Adds a script to check version in `CHANGELOG.md` against user specified files.

Same make targets are added, and the check is added to the make check target, and consequently becomes part of CI.

Files to be synced are now specified in the `Makefile` rather than hardcoded in the script.

`create_validation_csv_files.py` has also been updated to fetch the version dynamically for the default API path if no ENV variable is set.

Note I have set up some of files to be synced with the latest _release_ version instead of the latest version. You can see the details of that in the `Makefile` calls to `version-sync.sh`. Is that what we want? If not, leave me feedback and I'll correct that now.

#### Testing:
Basic checking of script:

1. Change the version in `preprocessing-pipeline-family.yaml` to an invalid version.
1. Run `make check-version`.
1. Should fail with an error message indicating valid semver could not be found in `preprocessing-pipeline-family.yaml`.
1. Run `make version-sync`.
1. Should fail with same error message.
1. Change the version in `preprocessing-pipeline-family.yaml` to a valid version, that is different from latest version in `CHANGELOG.md`.
1. Run `make check-version`.
1. Should exit 1, reporting diffs that would fix the version in `preprocessing-pipeline-family.yaml`.
1. Run `make version-sync`.
1. `preprocessing-pipeline-family.yaml` should now be in sync with `CHANGELOG.md`.
1. Open `CHANGELOG.md` and delete the contents of the file, and save it empty.
1. Run `make check-version`.
1. Should fail with an error message indicating valid semver could not be found in `CHANGELOG.md`.
1. Run `make version-sync`.
1. Should fail with same error message.
1. Undo changes in files.
1. `make check-version` and `make version-sync` should now exit 0 with no changes.

There's a lot of moving parts in this script as files to check can now be specified, and this doesn't really test everything. You might search the repo directory for the semver regex pattern to confirm the files I'm syncing.
